README.md

```diff
@@ -17,293 +17,18 @@ If you have a server-side Swift application or a cross-platform (e.g. Linux, mac
-
-In more complex cases, when `Signal`-trapping-based shutdown is not appropriate, you may pass `nil` as the `shutdownSignal` configuration, and call `shutdown` manually when appropriate. This is designed to be a rarely used pressure valve.
-
-`shutdown` is an asynchronous operation. Errors will be logged and bubbled up to the provided completion handler.
-
-### Stateful handlers
-
-In some cases it is useful to have the Start handlers return a state that can be passed on to the Shutdown handlers for shutdown.
-For example, when establishing some sort of a connection that needs to be closed at shutdown.
-
-```swift
-struct Foo {
-  func start() throws -> Connection {
-      return ...
-  }
-
-  func shutdown(state: Connection) throws {
-      ...
-  }
-}
-```
-
-```swift
-let foo = ...
-lifecycle.registerStateful(
-    label: "foo",
-    start: .sync(foo.start),
-    shutdown: .sync(foo.shutdown)
-)
-```
-
-### Complex Systems and Nesting of Subsystems
-
-In larger Applications (Services) `ComponentLifecycle` can be used to manage the lifecycle of subsystems, such that `ServiceLifecycle` can start and shutdown `ComponentLifecycle`s.
-
-In fact, since `ComponentLifecycle` conforms to `LifecycleTask`,
-it can start and stop other `ComponentLifecycle`s, forming a tree. E.g.:
-
-```swift
-struct SubSystem {
-    let lifecycle = ComponentLifecycle(label: "SubSystem")
-    let subsystem: SubSubSystem
-
-    init() {
-        self.subsystem = SubSubSystem()
-        self.lifecycle.register(self.subsystem.lifecycle)
-    }
-
-    struct SubSubSystem {
-        let lifecycle = ComponentLifecycle(label: "SubSubSystem")
-
-        init() {
-            self.lifecycle.register(...)
-        }
-    }
-}
-
-let lifecycle = ServiceLifecycle()
-let subsystem = SubSystem()
-lifecycle.register(subsystem.lifecycle)
-
-lifecycle.start { error in
-    ...
-}
-lifecycle.wait()
-```
-
-### Compatibility with SwiftNIO Futures
-
-[SwiftNIO](https://github.com/apple/swift-nio) is a popular networking library that among other things provides Future abstraction named `EventLoopFuture`.
-
-Swift Service Lifecycle comes with a compatibility module designed to make managing SwiftNIO based resources easy.
-
-Once you import `LifecycleNIOCompat` module, `LifecycleHandler` gains a static helper named `eventLoopFuture` designed to help simplify the registration call to:
-
-```swift
-let foo = ...
-lifecycle.register(
-    label: "foo",
-    start: .eventLoopFuture(foo.start),
-    shutdown: .eventLoopFuture(foo.shutdown)
-)
-```
-
-or, just shutdown:
-
-```swift
-let foo = ...
-lifecycle.registerShutdown(
-    label: "foo",
-    .eventLoopFuture(foo.shutdown)
-)
-```
-
--------
+You can find the documentation on how to use ServiceLifecycle over [here](https://swiftpackageindex.com/swift-server/swift-service-lifecycle/main/documentation/lifecycle).
```

Comment:
there is allot of value in providing key examples right on the readme. i suggest we do that, especially since  the full documentation is a hop away